### PR TITLE
[ARM plugin] Separate transformation module (for FP16 tests)

### PR DESCRIPTION
**Pull request by allnes**

### CODE REVIEW

**Comment by allnes** [comment]
@ilya-lavrenov please check

File: modules/arm_plugin/src/arm_executable_network.cpp [hunk 1]
```diff
@@ -9,14 +9,14 @@
...
 #include <vector>
 
-#include <ie_metric_helpers.hpp>
-#include <ie_plugin_config.hpp>
-#include <ie_ngraph_utils.hpp>
-#include <threading/ie_executor_manager.hpp>
-#include <ngraph/function.hpp>
-#include <ie_icore.hpp>
+#include "ie_metric_helpers.hpp"
+#include "ie/ie_plugin_config.hpp"
```
Comment (ilya-lavrenov) [inline modules/arm_plugin/src/arm_executable_network.cpp:13]: Why do we need to add `ie`? I suppose `openvino::runtime` target provides path to `ie_plugin_config.hpp` directly

File: modules/arm_plugin/src/transformations/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        ${CMAKE_CURRENT_SOURCE_DIR}/*.hpp)
+
+add_library(${TARGET_NAME} STATIC ${SOURCES_TRANSFORMATIONS})
+
+ie_arm_neon_optimization_flags(neon_flags)
+set_source_files_properties(${SOURCES_TRANSFORMATIONS} PROPERTIES COMPILE_OPTIONS "${neon_flags}")
+
+
+target_include_directories(${TARGET_NAME} PRIVATE
+        "${HEADERS_TRANSFORMATIONS}"
```
Comment (ilya-lavrenov) [inline modules/arm_plugin/src/transformations/CMakeLists.txt:19]: `HEADERS_TRANSFORMATIONS` is a list of headers, while we need to provide an include directory.
I suppose we need to have only `CMAKE_CURRENT_SOURCE_DIR` as an include dir

File: modules/arm_plugin/src/transformations/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+target_link_libraries(${TARGET_NAME} PRIVATE
+        arm_compute::arm_compute
+        arm_compute::arm_compute_core
+        arm_compute::half
+        openvino::ngraph_reference
+        openvino::util
+        openvino::itt
+        openvino::runtime::dev
+        openvino::runtime)
```
Comment (ilya-lavrenov) [inline modules/arm_plugin/src/transformations/CMakeLists.txt:30]: I suppose here we need only `openvino::runtime::dev`

File: modules/arm_plugin/src/arm_executable_network.cpp (hunk 1 above)
Comment (allnes) [reply modules/arm_plugin/src/arm_executable_network.cpp]: @ilya-lavrenov done

File: modules/arm_plugin/src/transformations/CMakeLists.txt (hunk 2 above)
Comment (allnes) [reply modules/arm_plugin/src/transformations/CMakeLists.txt]: @ilya-lavrenov done

File: modules/arm_plugin/src/transformations/CMakeLists.txt (hunk 3 above)
Comment (allnes) [reply modules/arm_plugin/src/transformations/CMakeLists.txt]: @ilya-lavrenov done

File: modules/arm_plugin/src/CMakeLists.txt [hunk 4]
```diff
@@ -21,16 +32,17 @@ ie_add_plugin(NAME ${TARGET_NAME}
               VERSION_DEFINES_FOR arm_plugin.cpp)
 
 target_include_directories(${TARGET_NAME} PRIVATE
-    "${CMAKE_CURRENT_SOURCE_DIR}"
-    "${IE_MAIN_ARM_PLUGIN_SOURCE_DIR}/include")
+        "${CMAKE_CURRENT_SOURCE_DIR}"
+        "${IE_MAIN_ARM_PLUGIN_SOURCE_DIR}/include"
+        "${HEADERS_TRANSFORMATIONS}")
```
Comment (ilya-lavrenov) [inline modules/arm_plugin/src/CMakeLists.txt:37]: HEADERS_TRANSFORMATIONS is not needed here

File: modules/arm_plugin/src/transformations/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+set(TARGET_NAME "openvino_arm_cpu_transformations")
+
+file(GLOB_RECURSE SOURCES_TRANSFORMATIONS
+        ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp)
+
+add_library(${TARGET_NAME} STATIC ${SOURCES_TRANSFORMATIONS})
+
+ie_arm_neon_optimization_flags(neon_flags)
+set_source_files_properties(${SOURCES_TRANSFORMATIONS} PROPERTIES COMPILE_OPTIONS "${neon_flags}")
```
Comment (ilya-lavrenov) [inline modules/arm_plugin/src/transformations/CMakeLists.txt:13]: I suppose NEON is not needed for transformations as well

File: modules/arm_plugin/src/CMakeLists.txt (hunk 4 above)
Comment (allnes) [reply modules/arm_plugin/src/CMakeLists.txt]: @ilya-lavrenov done

File: modules/arm_plugin/src/transformations/CMakeLists.txt (hunk 5 above)
Comment (allnes) [reply modules/arm_plugin/src/transformations/CMakeLists.txt]: @ilya-lavrenov done

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**